FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.cpp

```diff
@@ -8,27 +8,176 @@ Hdf5SummaryReader::Hdf5SummaryReader(const std::string file_path)
+    auto dataset_exists = H5Lexists(group.getId(), "GRIDPROPTIME", H5F_ACC_RDONLY);
+    // std::cout << "dataset_exists = " << dataset_exists << std::endl;
+
+    std::vector<std::vector<double>> sgas, soil, swat;
+
+    /*!
+    \todo Overall, this needs a better implementation, e.g., read soil and sgas
+    at the same time from the H5 group, instead of one vector at a time, which
+    appears inefficient... on the other hand, this reading needs to be made 
+    flexible/robust against the different phase combinations that might exist
+    in GRIDPROPTIME, e.g., soil/sgas, soil/sgas/swat, soil/swat, etc. 
+    */
+
+    /*!
+    \todo Later: to save space, load saturation data as additional columns in 
+    PTZ (replacing current compositional columns? comtrolled by custom KEYWORD?),
+    and remove GRIDPROPTIME completely
+    */
+
+    if (dataset_exists) {
+
+        // std::cout << "number of phases = " << number_of_phases() << std::endl;
+
+        hsize_t SOIL, SWAT, SGAS;
+        if (number_of_phases() < 3){
+            SGAS = 0;
+            SOIL = 2; // col: 3
+            SWAT = 1; // col: 2
+        }else{
+            SGAS = 1; // col: 2
+            SOIL = 2; // col: 3
+            SWAT = 3; // col: 4
+        }
+
+        sgas_ = getSaturation(group, SGAS);
+        soil_ = getSaturation(group, SOIL);
+        swat_ = getSaturation(group, SWAT);
+
+    }else{
+
+        sgas_ = reservoir_pressure();
+        soil_ = reservoir_pressure();
+        swat_ = reservoir_pressure();
+    }
+}
+
+std::vector<std::vector<double>> Hdf5SummaryReader::getSaturation(Group group, hsize_t sat_type) {
+
+    // Read the file
+    DataSet dataset = DataSet(group.openDataSet(DATASET_NAME_SATURATION));
+    DataSpace dataspace = dataset.getSpace();
+    hsize_t dims[3];
+    dataspace.getSimpleExtentDims(dims, NULL);
+
+    std::vector<double> vector;
+    vector.resize(dims[0] * dims[2]);
+
+    // std::cout << "sat_type = " << (int)sat_type << std::endl;
+    std::vector<std::vector<double>>  sat_; //!< Temporary saturation vector.
+
+    if (sat_type > 0){
+        // Define hyperslab
+        hsize_t count[3] = {dims[0], 1, dims[2]};
+        hsize_t offset[3] = {0, sat_type, 0};
+        dataspace.selectHyperslab(H5S_SELECT_SET, count, offset);
+
+        // Size of selected column + define memory space
+        hsize_t col_sz[2] = {dims[0], dims[2]};
+        DataSpace mspace(2, col_sz);
+
+        dataset.read(vector.data(), PredType::NATIVE_DOUBLE, mspace, dataspace);
+    }else{
+        vector.resize(dims[0] * dims[2],0);
+    }
+
+    int ncells = (int)dims[0];
+    int ntime_steps = (int)dims[2];
+
+    for (int tt = 0; tt < ntime_steps; ++tt) {
+        std::vector<double> sub;
+        for (int cc = 0; cc < ncells; ++cc) {
+            sub.push_back(vector[ cc * ntime_steps + tt ]);
+        }
+        sat_.push_back(sub);
+    }
+
+    return sat_;
+}
+
+void Hdf5SummaryReader::readReservoirPressure(std::string file_path) {
```

Comment:
Is there a reason the pattern for reading from HDF5 files used in the existing methods (for reading well rates) hasn't been used in this function?